allennlp/data/dataset_readers/dataset_utils/span_utils.py

```diff
@@ -0,0 +1,125 @@
+from typing import List, Tuple, Callable
+
+from allennlp.data.dataset_readers.dataset_utils.ontonotes import TypedStringSpan
+from allennlp.data.tokenizers.token import Token
+
+def enumerate_spans(sentence: List[Token],
+                    offset: int = 0,
+                    max_span_width: int = None,
+                    min_span_width: int = None,
+                    filter_function: Callable[[List[Token]], bool] = None) -> List[Tuple[int, int]]:
+    """
+    Given a sentence, return all token spans within the sentence. Spans are `inclusive`.
+    Additionally, you can provide a maximum and minimum span width, which will be used
+    to exclude spans outside of this range.
+
+    Finally, you can provide a function mapping ``List[Token] -> bool``, which will
+    be applied to every span to decide whether that span should be included. This
+    allows filtering by length, regex matches, pos tags or any Spacy ``Token`` attribute.
+
+    Parameters
+    ----------
+    sentence : ``List[Token]``, required.
+        The sentence to generate spans for.
+    offset : ``int``, optional (default = 0)
+        A numeric offset to add to all span start and end indices. This is helpful
+        if the sentence is part of a larger structure, such as a document, which
+        the indices need to respect.
+    max_span_width : ``int``, optional (default = None)
+        The maximum length of spans which should be included.
+    min_span_width : ``int``, optional (default = None)
+        The minimum length of spans which should be included.
+    """
+    max_span_width = max_span_width or len(sentence)
+    min_span_width = min_span_width or 0
+    filter_function = filter_function or bool
```

Comment:
`bool` here seems a bit confusing.  This actually interacts in funny ways with `min_span_width = 0` - `bool` gives `False` for that (at least, I thought it should, be your test seems to say otherwise...  not sure what's going on there).  `lambda x: True` would be better.

Reply:
The problem is you can't assign a lambda function in an `or` statement, which seems to be a weird python quirk....